src/set.ts

```diff
@@ -217,4 +217,23 @@ export class SuperchargedSet<T> implements Iterable<T> {
   toArray (): T[] {
     return Array.from(this.set)
   }
+
+  /**
+   * Returns the number of items matching the given predicate.
+   *
+   * @param {Function} predicate
+   *
+   * @returns {number}
+   */
+  count (predicate: (item: T) => boolean): number {
+    let count = 0
+
+    for (const value of this.set.values()) {

```

Comment:
The set already has the `.filter(predicate)` and `.size()` methods. I would like to see the `.count(predicate)` using both. What do you think?

Reply:
I think your approach of reusing methods is better. I have submit a new refractor commit.